avocado/utils/disk.py

```diff
@@ -93,3 +93,48 @@ def get_filesystem_type(mount_point='/'):
             _, fs_file, fs_vfstype, _, _, _ = mount_line.split()
             if fs_file == mount_point:
                 return fs_vfstype
+
+
+def get_partition_info(device):
+    """
+    Get partition info
+    :param device: the device, e.g. /dev/sda3
+    :return: dictionary which has all partition info
+    ## Todo
+    #right now this utility works for block device
+    #still nvme kind of device support need to be added
+    """
+    if device in get_disks():
+        raise DiskUtilsError('provided disk partition not disk')
+
+    values = list()
+
+    disk_device = device.rstrip('0123456789')
+    # Parse fdisk output to get partition info.  Ugly but it works.
+
+    fdisk_fd = os.popen("/sbin/fdisk -l -u  '%s'" % disk_device)
```

Comment:
It's sometimes hard to find balance between writing new code, and using an existing tool.  In this specific situation, I think should evaluate the former option.  At least in the case of MBRs, it's pretty easy to retrieve information on the partitions.  The following Python code is a good example:

https://github.com/CdtDelta/mbr_parse/blob/master/mbr_parse.py

The amount of code (~ 100 lines) is really small (not far from the amount of code here).  We can even ask the author for permission to use the code.

Reply:
@clebergnu sure make sense , can you please help me how we can proceed  with this so we can reuse his same code with permission of author having mentioned his authorship for that particular portion of code .